src/Definition/Oracle.php

```diff
@@ -0,0 +1,231 @@
+                    $valid = false;
+                    break;
+                }
+                if ((int) $length <= 0 || (int) $length > 2147483647) {
+                    $valid = false;
+                    break;
+                }
+                break;
+            case "DATE":
+                if (is_null($length) || $length === "") {
+                    break;
+                }
+                if (!is_numeric($length)) {
+                    $valid = false;
+                    break;
+                }
+                if ((int) $length < 0 || (int) $length > 6) {
+                    $valid = false;
+                    break;
+                }
+                break;
+            case "NUMBER":
+                if (is_null($length) || $length === "") {
+                    break;
+                }
+                $parts = explode(",", $length);
+                if (count($parts) > 2 || count($parts) < 1) {
+                    $valid = false;
+                    break;
+                }
+                if (!is_numeric($parts[0])) {
+                    $valid = false;
+                    break;
+                }
+                if (isset($parts[1]) && !is_numeric($parts[1])) {
+                    $valid = false;
+                    break;
+                }
+                if ((int) $parts[0] <= 0 || (int) $parts[0] > 255) {
+                    $valid = false;
+                    break;
+                }
+                break;
+
+            case "DECIMAL":
+            case "DEC":
+            case "FIXED":
+            case "NUMERIC":
+                if (is_null($length) || $length == "") {
+                    break;
+                }
+                $parts = explode(",", $length);
+                if (count($parts) > 2 || count($parts) < 1) {
+                    $valid = false;
+                    break;
+                }
+                if (!is_numeric($parts[0])) {
+                    $valid = false;
+                    break;
+                }
+                if (isset($parts[1]) && !is_numeric($parts[1])) {
+                    $valid = false;
+                    break;
+                }
+                if ((int)$parts[0] <= 0 || (int)$parts[0] > 65) {
+                    $valid = false;
+                    break;
+                }
+                if (isset($parts[1]) && ((int)$parts[1] > (int)$parts[0] || (int)$parts[1] > 30)) {
+                    $valid = false;
+                    break;
+                }
+                break;
+            default:
+                if (!is_null($length) && $length != "") {
+                    $valid = false;
+                    break;
+                }
+                break;
+        }
+        if (!$valid) {
+            throw new InvalidLengthException("'{$length}' is not valid length for {$type}");
+        }
+    }
+
+    /**
+     * @return string
+     */
+    public function getBasetype()
```

Comment:
This method needs to accept any type and return a matching basetype.  Missing here are mappings to `FLOAT` `INTEGER`, `BOOLEAN` and `TIMESTAMP` datatypes.